express-notes-api/index.js

```diff
@@ -0,0 +1,165 @@
+const express = require('express');
+const fs = require('fs');
+
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/notes', (req, res) => {
+  const data = require('./data.json');
+  const arr = [];
+
+  for (const entry in data.notes) {
+    arr.push(data.notes[entry]);
+  };
+  res.status(200).send(arr);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const data = require('./data.json');
+  const id = req.params.id;
+  let found = false;
+
+  if (id >= 1) {
+    for (const entry in data.notes) {
+      if (entry === id) {
+        const resJSON = data.notes[entry];
+        res.status(200).json(resJSON);
+        found = true;
+      };
+    };
+    if (!found) {
+      const errorObj = {error: `cannot find note with id ${id}`};
+      res.status(404).json(errorObj);
+    };
+  } else {
+    const errorObj = {error: 'id must be a positive integer'};
+    res.status(400).json(errorObj);
+  };
+});
+
+app.post('/api/notes', (req, res) => {
+  const data = require('./data.json');
+  const inputObj = req.body;
+  const errorObj = {error: 'content is a required field'};
+
+  if (Object.keys(inputObj).length === 0 || inputObj.content === undefined) {
+    res.status(400).json(errorObj);
+  } else {
+    let nextIdValue = data.nextId;
+
+    data.notes[nextIdValue] = {id: '', content: ''};
+    data.notes[nextIdValue].id = nextIdValue;
+    data.notes[nextIdValue].content = inputObj.content;
+    data.nextId++;
+
+    let dataJSON = JSON.stringify(data, null, 2);
+
+    const errFunc = (err) => {
+      if (err) {
+        errorObj.error = 'An unexpected error occurred.';
+        res.status(500).json(errorObj);
+     } else {
+        res.status(201).json(data.notes[nextIdValue]);
+      }
+    };
+    fs.writeFile('data.json', dataJSON, 'utf8', errFunc);
+  };
+});
+
+app.delete('/api/notes/', (req, res) => {
+  const errorObj = {error: 'id must be a postivie integer'};
+  res.status(400).json(errorObj);
+});
```

Comment:
while this is a nice thought, you can perform this error handling in the delete route below